src/sonic-yang-models/yang-models/sonic-lossless-traffic-pattern.yang

```diff
@@ -0,0 +1,58 @@
+module sonic-lossless-traffic-pattern {
+
+    yang-version 1.1;
+
+    namespace  "http://github.com/Azure/sonic-lossless-traffic-pattern";
+
+    prefix lossless-traffic-pattern;
+
+    description "LOSSLESS_TRAFFIC_PATTERN YANG module for SONiC OS";
+
+    revision 2022-05-31 {
+        description "Initial version";
+    }
+
+    container sonic-lossless-traffic-pattern {
+        container LOSSLESS_TRAFFIC_PATTERN {
+
+            description "LOSSLESS_TRAFFIC_PATTERN part of config_db.json";
+
+            list LOSSLESS_TRAFFIC_PATTERN_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})' {
```

Comment:
Do we restrict the name in code with this pattern today? If not I prefer not to introduce any new patterns here as it cause breaks with already configured names which don't align with this pattern or length

Reply:
No checking in code today but this is to align other QoS tables like [DSCP_TO_TC_MAP](https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-dscp-tc-map.yang#L35)

Reply:
In this case I prefer not to impose any restrictions that are not present today unless it adds some value or functionality.

Reply:
As discussed offline, let's keep the pattern checking to follow the current practice in QoS and buffer part.